### PR TITLE
tests/str: Rework sysinstall-tests to be an entrypoint

### PR DESCRIPTION
**Pull request by cgwalters**
Prep for creating more types of tests.

Move copying of `tests/` into the sysinstall-tests rather than `overlay-git`
as not all test types may need that.

Factored out of https://github.com/ostreedev/ostree/pull/1501

### CODE REVIEW

**Comment by cgwalters** [comment]
OK, some more patches here.

File: tests/installed/itest-payload-link.sh [hunk 1]
```diff
@@ -57,29 +56,25 @@ fi
 mkfs.xfs -m reflink=1 ${blkdev}
 
 mount ${blkdev} mnt
+cd mnt
 
-test_tmpdir=$(pwd)/mnt
-cd ${test_tmpdir}
-
+# Test that coreutils will do reflink
```
Comment (cgwalters) [inline tests/installed/itest-payload-link.sh]: Also I changed this to just require that reflinks work since that's the point of the test.

**Comment by cgwalters** [comment]
:green_heart:

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 263cb35

**Comment by jlebon** [comment]
(Also, thanks for splitting this up into smaller digestible commits!)

**Pull request closed without merging**